dipy/viz/ui.py

```diff
@@ -0,0 +1,612 @@
+        callback : function
+            The callback function.
+        priority : int
+        """
+        # Actually since we need an interactor style we will add the callback
+        # only when this UI component is added to the renderer.
+        self._callbacks.append((prop, event_type, callback, priority))
+
+    def set_center(self, position):
+        """ Sets the center of the UI component
+
+        Parameters
+        ----------
+        position : (float, float)
+        """
+        msg = "Subclasses of UI must implement `set_center(self, position)`."
+        raise NotImplementedError(msg)
+
+    def set_visibility(self, visibility):
+        """ Sets visibility of this UI component and all its sub-components. """
+        for actor in self.get_actors():
+            actor.SetVisibility(visibility)
+
+    def handle_events(self):
+        self.add_callback("LeftButtonPressEvent", self.left_button_click_callback)
+        self.add_callback("LeftButtonReleaseEvent", self.left_button_release_callback)
+        self.add_callback("RightButtonPressEvent", self.right_button_click_callback)
+        self.add_callback("RightButtonReleaseEvent", self.right_button_release_callback)
+        self.add_callback("MouseMoveEvent", self.mouse_move_callback)
+
+    @staticmethod
+    def left_button_click_callback(i_ren, obj, self):
+        self.left_button_state = "clicked"
+        i_ren.event.abort()
+
+    @staticmethod
+    def left_button_release_callback(i_ren, obj, self):
+        if self.left_button_state == "clicked":
+            self.on_left_mouse_button_pressed(i_ren, obj, self)
+        self.left_button_state = "released"
+
+    @staticmethod
+    def right_button_click_callback(i_ren, obj, self):
+        self.right_button_state = "clicked"
+        i_ren.event.abort()
+
+    @staticmethod
+    def right_button_release_callback(i_ren, obj, self):
+        if self.right_button_state == "clicked":
+            self.on_right_mouse_button_pressed(i_ren, obj, self)
+        self.right_button_state = "released"
+
+    @staticmethod
+    def mouse_move_callback(i_ren, obj, self):
+        if self.left_button_state == "clicked" or self.left_button_state == "dragging":
+            self.left_button_state = "dragging"
+            self.on_left_mouse_button_drag(i_ren, obj, self)
+        elif self.right_button_state == "clicked" or self.right_button_state == "dragging":
+            self.right_button_state = "dragging"
+            self.on_right_mouse_button_drag(i_ren, obj, self)
+        else:
+            pass
+
+
+class Button2D(UI):
+    """A 2D overlay button and is of type vtkTexturedActor2D.
+    Currently supports:
+    - Multiple icons.
+    - Switching between icons.
+
+    Attributes
+    ----------
+    size: (float, float)
+        Button Size.
+    """
+
+    def __init__(self, icon_fnames, size=(30, 30)):
+        """
+        Parameters
+        ----------
+        size : 2-tuple of int, optional
```

Comment:
@MarcCote I can see `tuple (2,)` format used at places and at some other places `(type, type)` is used as we have done.